identity-service/src/routes/trackListens.js

```diff
@@ -233,68 +232,67 @@ const initializeExpiringRedisKey = async (redis, key, expiry) => {
   }
 }
 
+const TRACKING_LISTEN_SUBMISSION_KEY = 'listens-tx-submission-ts'
+const TRACKING_LISTEN_SUCCESS_KEY = 'listens-tx-success-ts'
+
 module.exports = function (app) {
   app.get('/tracks/listen/solana/status', handleResponse(async (req, res) => {
     const redis = req.app.get('redis')
     const results = await redis.keys('listens-tx-*')
     // Expected percent success
     let { percent = 0.9, cutoffMinutes = 60 } = req.query
     let hourlyResponseData = {}
-    let success = 0
-    let submission = 0
     // Example key format = listens-tx-success::2022-01-25T21:00:00.000Z
     for (var entry of results) {
       let split = entry.split('::')
       if (split.length >= 2) {
         let hourSuffix = split[1]
-        let trackingRedisKeys = getTrackingListenKeys(hourSuffix)
+        const trackingRedisKeys = getTrackingListenKeys(hourSuffix)
 
         if (!hourlyResponseData.hasOwnProperty(hourSuffix)) {
           hourlyResponseData[hourSuffix] = {
             submission: Number(await redis.get(trackingRedisKeys.submission)),
             success: Number(await redis.get(trackingRedisKeys.success)),
             time: new Date(hourSuffix)
           }
-          submission += hourlyResponseData[hourSuffix].submission
-          success += hourlyResponseData[hourSuffix].success
         }
       }
     }
 
-    const percentSuccess = success / submission
+    // Clean up time series entries that are greater than 1 week old
+    const oldestExpireMillis = Date.now() - redisTxTrackingExpirySeconds * 1000
+    await redis.zremrangebyscore(TRACKING_LISTEN_SUBMISSION_KEY, 0, oldestExpireMillis)
+    await redis.zremrangebyscore(TRACKING_LISTEN_SUCCESS_KEY, 0, oldestExpireMillis)
+
+    const totalSuccessCount = await redis.zcount(TRACKING_LISTEN_SUCCESS_KEY, 0, Number.MAX_SAFE_INTEGER)
+    const totalSubmissionCount = await redis.zcount(TRACKING_LISTEN_SUBMISSION_KEY, 0, Number.MAX_SAFE_INTEGER)
+    const totalPercentSuccess = totalSuccessCount / totalSubmissionCount
     // Sort response in descending time order
     const sortedHourlyData =
       Object.keys(hourlyResponseData)
         .sort((a, b) => (new Date(b) - new Date(a)))
         .map(key => hourlyResponseData[key])
 
     // Calculate success of submissions before the cutoff
-    let recentSubmissionCount = 0
-    let recentSuccessCount = 0
-    let cutoffTimestamp
     const now = Date.now()
-
-    for (var sortedHourlyEntry of sortedHourlyData) {
-      // Convert diff from ms to minutes
-      const diff = (now - sortedHourlyEntry.time.getTime()) / 60000
-      recentSubmissionCount += sortedHourlyEntry.submission
-      recentSuccessCount += sortedHourlyEntry.success
-      cutoffTimestamp = sortedHourlyEntry.time
-      // Exit calculation
-      if (diff > cutoffMinutes) {
-        break
-      }
-    }
-
+    const nowPlusEntropy = now + 9 // Account for the fact that each date has a random UUID appended to it
```

Comment:
not super clear how adding 9 accounts for a random UUID? is this intended to add more entropy? bc it will be predictable if we do it for every listen

Reply:
ahhh this makes sense

Reply:
For example, if the current time in millis is `00000000` and the entropy is `b2192362-4b19-4f5c-ba5f-c3cea0ee3208` then the redis entry is `0000000b2192362-4b19-4f5c-ba5f-c3cea0ee3208`

So a redis query for [0, `00000000`] wouldn't include the listen for the current time since `00000000` < `0000000b2192362-4b19-4f5c-ba5f-c3cea0ee3208`. But a query for [0, `00000001`] would include the current time

Reply:
Ok cool, merging! Thanks!